tests/cmd/cmd_test.go

```diff
@@ -114,3 +115,37 @@ func Test_builderror(t *testing.T) {
 		t.Errorf("Test Failed")
 	}
 }
+
+func Test_examples(t *testing.T) {
+	// Test Examples from `examples/ directory`
+	files, err := ioutil.ReadDir("../../examples")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, f := range files {
+		cmdStr := fmt.Sprintf("%s generate -f %s", BinaryLocation, ProjectPath+"examples/"+f.Name())
+		output, err := exec.Command("/bin/sh", "-c", cmdStr).Output()
+		if err != nil {
+			t.Errorf("kedge generate failed for - %s\n Error is - %s", f.Name(), output)
+		}
+	}
+
+	// Test Examples from `docs/examples/ directory`
+	files, err = ioutil.ReadDir("../../docs/examples")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, f := range files {
+		if f.Name() == "README.md" || f.Name() == "build" || f.Name() == "s2i" {
```

Comment:
Why is whole s2i dir skipped here? There kedge files in it which should be tested.

It might be better to just include all `*.yml` and `*.yaml` files instead of excluding the others.


Also, it might be better to use [filepath.Walk](
https://golang.org/pkg/path/filepath/#Walk) to get all the files (even those nested deep in the file structure)